src/vector/platform/VectorBasePlatform.ts

```diff
@@ -37,14 +35,9 @@ export const updateCheckStatusEnum = {
 /**
  * Vector-specific extensions to the BasePlatform template
  */
-export default class VectorBasePlatform extends BasePlatform {
-    constructor() {
-        super();
-
-        this.showUpdateCheck = false;
-        this.startUpdateCheck = this.startUpdateCheck.bind(this);
-        this.stopUpdateCheck = this.stopUpdateCheck.bind(this);
-    }
+export default abstract class VectorBasePlatform extends BasePlatform {
+    protected showUpdateCheck: boolean = false;
+    protected _favicon: Favicon;
```

Comment:
Is there a specific style convention for `_` prefix vs. not in TS land for protected / private fields? I noticed you removed the `_` for some private fields, but these protected ones use a mix, so wondering if there's a particular style to follow.

Reply:
favicon is already used by a getter so conflicted, otherwise I would have had no prefix here either.

Prefix normally denotes non-public but TS gives us protected/private to mark them as such explicitly.

Reply:
typically the workaround is to name it something different like `faviconInternal` or `rawFavicon` or `faviconDontUseThis` or `faviconWhyAreYouAccessingThisUseTheSetterAndGetter`

Reply:
> faviconWhyAreYouAccessingThisUseTheSetterAndGetter

hah

Reply:
Ah right, the getter. Okay, well, I don't have any particularly great suggestions, any of @turt2live's or leaving as-is seem okay to me.